File: src/test/scala/com/kjetland/jackson/jsonSchema/testDataScala/ClassUsingValidation.scala
package com.kjetland.jackson.jsonSchema.testDataScala

import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaInject
import javax.validation.constraints._
import javax.validation.groups.Default

case class ClassUsingValidation
(
  @NotNull
  stringUsingNotNull:String,

  @NotBlank
  stringUsingNotBlank:String,

  @NotNull
  @NotBlank
  stringUsingNotBlankAndNotNull:String,

  @NotEmpty
  stringUsingNotEmpty:String,

  @NotEmpty
  notEmptyStringArray:List[String], // Per PojoArraysWithScala, we use always use Lists in Scala, and never raw arrays.

  @NotEmpty
  notEmptyMap:Map[String, String],

  @Size(min=1, max=20)
  stringUsingSize:String,

  @Size(min=1)
  stringUsingSizeOnlyMin:String,

  @Size(max=30)
  stringUsingSizeOnlyMax:String,

  @Pattern(regexp = "_stringUsingPatternA|_stringUsingPatternB")
  stringUsingPattern:String,

  @Pattern.List(Array(
    new Pattern(regexp = "^_stringUsing.*"),
    new Pattern(regexp = ".*PatternList$")
  ))
  stringUsingPatternList:String,

  @Min(1)
  intMin:Int,
  @Max(10)
  intMax:Int,
  @Min(1)
  doubleMin:Double,
  @Max(10)
  doubleMax:Double,
  @DecimalMin("1.5")
  decimalMin:Double,
  @DecimalMax("2.5")
  decimalMax:Double,

  @Email
  email:String
)

trait ValidationGroup1
trait ValidationGroup2
trait ValidationGroup3_notInUse

@JsonSchemaInject(json = """{"injected":true}""", javaxValidationGroups = Array(classOf[ValidationGroup1]))
case class ClassUsingValidationWithGroups
(
  @NotNull
  @JsonSchemaInject(json = """{"injected":true}""")
  noGroup:String,

  @NotNull(groups = Array(classOf[Default]))
  @JsonSchemaInject(json = """{"injected":true}""", javaxValidationGroups = Array(classOf[Default]))
  defaultGroup:String,

  @NotNull(groups = Array(classOf[ValidationGroup1]))
  @JsonSchemaInject(json = """{"injected":true}""", javaxValidationGroups = Array(classOf[ValidationGroup1]))
  group1:String,

  @NotNull(groups = Array(classOf[ValidationGroup2]))
  @JsonSchemaInject(json = """{"injected":true}""", javaxValidationGroups = Array(classOf[ValidationGroup2]))
  group2:String,

  @NotNull(groups = Array(classOf[ValidationGroup1], classOf[ValidationGroup2]))
  @JsonSchemaInject(json = """{"injected":true}""", javaxValidationGroups = Array(classOf[ValidationGroup1], classOf[ValidationGroup2]))
  group12:String

)

File: src/test/java/com/kjetland/jackson/jsonSchema/testData/ClassUsingValidation.java
package com.kjetland.jackson.jsonSchema.testData;

import java.util.List;
import java.util.Map;
import javax.validation.constraints.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.experimental.FieldDefaults;
import lombok.extern.jackson.Jacksonized;

@FieldDefaults(level = AccessLevel.PUBLIC, makeFinal = true) @Jacksonized @Builder @EqualsAndHashCode // Can be Java 17 record
public class ClassUsingValidation
{
  @NotNull
  String stringUsingNotNull;

  @NotBlank
  String stringUsingNotBlank;

  @NotNull
  @NotBlank
  String stringUsingNotBlankAndNotNull;

  @NotEmpty
  String stringUsingNotEmpty;

  @NotEmpty
  List<String> notEmptyStringArray; // Per PojoArraysWithScala, we use always use Lists in Scala, and never raw arrays.

  @NotEmpty
  Map<String, String> notEmptyMap;

  @Size(min=1, max=20)
  String stringUsingSize;

  @Size(min=1)
  String stringUsingSizeOnlyMin;

  @Size(max=30)
  String stringUsingSizeOnlyMax;

  @Pattern(regexp = "_stringUsingPatternA|_stringUsingPatternB")
  String stringUsingPattern;

  @Pattern.List({
    @Pattern(regexp = "^_stringUsing.*"),
    @Pattern(regexp = ".*PatternList$")
  })
  String stringUsingPatternList;

  @Min(1)
  int intMin;
  @Max(10)
  int intMax;
  @Min(1)
  double doubleMin;
  @Max(10)
  double doubleMax;
  @DecimalMin("1.5")
  double decimalMin;
  @DecimalMax("2.5")
  double decimalMax;

  @Email
  String email;
}